webrtc/RTCConfiguration-iceCandidatePoolSize.html

```diff
@@ -1,35 +1,87 @@
+
+  ...
+
+  iceCandidatePoolSize of type octet, defaulting to 0
+    Size of the prefetched ICE pool as defined in [JSEP] (section 3.5.4. and section 4.1.1.).
+-->
 <script src="/resources/testharness.js"></script>
 <script src="/resources/testharnessreport.js"></script>
-</head>
-<body>
 <script>
-function testValidPoolSize(size, expectedSize)
-{
-    test(function() {
-        var pc = new RTCPeerConnection({iceCandidatePoolSize: size})
-        assert_equals(pc.getConfiguration().iceCandidatePoolSize, expectedSize !== undefined ? expectedSize : size)
-    }, "Setting iceCandidatePoolSize to a valid value: " + size)
-}
-
-function testInvalidPoolSize(size)
-{
-    test(function() {
-        assert_throws(new TypeError(), () => new RTCPeerConnection({iceCandidatePoolSize: size}))
-    }, "Setting iceCandidatePoolSize to an invalid value: " + size)
-}
-
-testValidPoolSize(10)
-testValidPoolSize(0)
-testValidPoolSize(null, 0)
-testValidPoolSize(undefined, 0)
-
-testInvalidPoolSize(-1)
-testInvalidPoolSize(1000)
-testInvalidPoolSize("string")
+  function initializeWithValidPoolSize(iceCandidatePoolSize, expectedSize) {
+    test(() => {
+      const pc = new RTCPeerConnection({
+        iceCandidatePoolSize
+      });
+      assert_equals(pc.getConfiguration().iceCandidatePoolSize, expectedSize);
+
+    }, `new RTCPeerConnection({ iceCandidatePoolSize }) with valid value: ${iceCandidatePoolSize}`);
+  }
+
+  function initializeWithInvalidPoolSize(iceCandidatePoolSize) {
+    test(() => {
+      assert_throws(new TypeError(), () => {
+        new RTCPeerConnection({
+          iceCandidatePoolSize
+        });
+      });
+    }, `new RTCPeerConnection({ iceCandidatePoolSize }) with invalid value: ${iceCandidatePoolSize}`);
+  }
+
+  function reconfigureWithValidPoolSize(iceCandidatePoolSize) {
+    test(() => {
+      const pc = new RTCPeerConnection();
+      pc.setConfiguration({ iceCandidatePoolSize });
+      assert_equals(pc.getConfiguration().iceCandidatePoolSize, expectedSize);
+
+    }, `setConfiguration({ iceCandidatePoolSize }) to a valid value: ${iceCandidatePoolSize}`);
+  }
+
+  function reconfigureWithInvalidPoolSize(iceCandidatePoolSize) {
+    test(() => {
+      const pc = new RTCPeerConnection();
+      assert_throws(new TypeError(), () => {
+        pc.setConfiguration({ iceCandidatePoolSize });
+      });
+    }, `setConfiguration({ iceCandidatePoolSize }) to an invalid value: ${iceCandidatePoolSize}`);
+  }
+
+  /*
+
+  dictionary RTCConfiguration {
+      ...
+      [EnforceRange]
+      octet                    iceCandidatePoolSize = 0;
+  };
+
+  ... of type octet
+  */
+  initializeWithValidPoolSize(0, 0);
+  initializeWithValidPoolSize(127, 127);
+  initializeWithValidPoolSize(255, 255);
+
+  // Defaulting to 0
+  initializeWithValidPoolSize(null, 0);
+  initializeWithValidPoolSize(undefined, 0);
```

Comment:
FWIW, I think it's fine to test things that are part of Web IDL, because often it can't be tested by idlharness.js because it can't assume anything about what getters and setters should do.